File: 2570.Merge-Two-2D-Arrays-by-Summing-Values.js
/**
 * @param {number[][]} nums1
 * @param {number[][]} nums2
 * @return {number[][]}
 */
var mergeArrays = function (nums1, nums2) {
  let idx1 = 0;
  let idx2 = 0;
  const result = [];

  while (idx1 < nums1.length && idx2 < nums2.length) {
    const [i1, v1] = nums1[idx1];
    const [i2, v2] = nums2[idx2];

    if (i1 === i2) {
      result.push([i1, v1 + v2]);
      idx1 += 1;
      idx2 += 1;
    } else if (i1 < i2) {
      result.push([i1, v1]);
      idx1 += 1;
    } else if (i1 > i2) {
      result.push([i2, v2]);
      idx2 += 1;
    }
  }

  while (idx1 < nums1.length) {
    result.push(nums1[idx1]);
    idx1 += 1;
  }

  while (idx2 < nums2.length) {
    result.push(nums2[idx2]);
    idx2 += 1;
  }

  return result;
};

File: python/2570.Merge-Two-2D-Arrays-by-Summing-Values.py
class Solution:
    def mergeArrays(self, nums1: List[List[int]], nums2: List[List[int]]) -> List[List[int]]:
        idx1, idx2 = 0, 0
        result = []

        while idx1 < len(nums1) and idx2 < len(nums2):
            i1, v1 = nums1[idx1]
            i2, v2 = nums2[idx2]

            if i1 == i2:
                result.append([i1, v1 + v2])
                idx1 += 1
                idx2 += 1
            elif i1 < i2:
                result.append(nums1[idx1])
                idx1 += 1
            else:
                result.append(nums2[idx2])
                idx2 += 1

        while idx1 < len(nums1):
            result.append(nums1[idx1])
            idx1 += 1

        while idx2 < len(nums2):
            result.append(nums2[idx2])
            idx2 += 1

        return result